### PR TITLE
adding benchmarking

### PR DESCRIPTION
**Pull request by adamomainz**
Summary:
adding benchmarking by pulling out all the parts of torch we want to replace with triton and adding them to a single operators file. I also had to make some changes to the existing kernels since they had relative paths to triton itself
 
Test Plan:
made sure I didnt break anything so far

`python3.9 -m main llama_chat_completion --profile=False --benchmark=True --ckpt_dir="models/llama/meta-llama/Meta-Llama-3-8B-Instruct/original" --tokenizer_path="models/llama/meta-llama/Meta-Llama-3-8B-Instruct/original/tokenizer.model" --use_triton=False`

Output from benchmarks:

Screenshot 2024-08-09 at 1 18 32 PM

### CODE REVIEW

File: models/llama/llama/mathOps.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
+import torch.nn.functional as F
```
Comment (bertmaher) [inline models/llama/llama/mathOps.py]: This file should be `math_ops.py` to keep with standard conventions

**Review by bertmaher — CHANGES_REQUESTED** [review]
What are the `False True` args in your example command line doing?  Let's make those proper named args so they're not a mystery :)

File: benchmarking/profiler.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                cls._instance.function_stack.append(record_name or func.__name__)
+                name = ".".join(cls._instance.function_stack)
+                if cls._instance.profiler and not skip_profiling:
+                    cls._instance.profiler.start()
+                start_time = time.perf_counter()
+                
+                with torch.profiler.record_function(name):
+                    result = func(*args, **kwargs)
+                
+                end_time = time.perf_counter()
```
Comment (bertmaher) [inline benchmarking/profiler.py]: `perf_counter` isn't the right way to measure GPU time.  Use `torch.cuda.Event` instead:
https://discuss.pytorch.org/t/how-to-measure-time-in-pytorch/26964

**Comment by adamomainz** [comment]
> What are the `False True` args in your example command line doing? Let's make those proper named args so they're not a mystery :)

@bertmaher fixed! was only in the test plan the params are properly named in the main.py file :)

File: models/llama/llama/mathOps.py (hunk 1 above)
Comment (adamomainz) [reply models/llama/llama/mathOps.py]: fixing

**Review by bertmaher — APPROVED** [review]

File: benchmarking/profiler.py (hunk 2 above)
Comment (adamomainz) [reply benchmarking/profiler.py]: https://github.com/triton-lang/kernels/issues/5

**Pull request merged**